test/unit/org/apache/cassandra/cql3/RandomSchemaTest.java

```diff
@@ -0,0 +1,204 @@
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.cassandra.config.CassandraRelevantProperties;
+import org.apache.cassandra.db.marshal.DurationType;
+import org.apache.cassandra.db.marshal.UserType;
+import org.apache.cassandra.schema.ColumnMetadata;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.utils.AbstractTypeGenerators;
+import org.apache.cassandra.utils.AbstractTypeGenerators.TypeGenBuilder;
+import org.apache.cassandra.utils.CassandraGenerators;
+import org.apache.cassandra.utils.CassandraGenerators.TableMetadataBuilder;
+import org.apache.cassandra.utils.FailingConsumer;
+import org.quicktheories.core.Gen;
+import org.quicktheories.core.RandomnessSource;
+import org.quicktheories.generators.SourceDSL;
+import org.quicktheories.impl.JavaRandom;
+
+public class RandomSchemaTest extends CQLTester.InMemory
+{
+    private static final Logger logger = LoggerFactory.getLogger(RandomSchemaTest.class);
+
+    static
+    {
+        // make sure blob is always the same
+        CassandraRelevantProperties.TEST_BLOB_SHARED_SEED.setInt(42);
+    }
+
+    @Test
+    public void test()
+    {
+        Gen<Boolean> nulls = SourceDSL.integers().between(1, 100).map(i -> i < 5);
+        qt().checkAssert(random -> {
+            TypeGenBuilder nonEmptyNoDuration = AbstractTypeGenerators.builder()
+                                                                      .withoutEmpty()
+                                                                      .withUserTypeKeyspace(KEYSPACE)
+                                                                      .withoutPrimitive(DurationType.instance);
+            TableMetadata metadata = new TableMetadataBuilder()
+                                     .withKeyspaceName(KEYSPACE)
+                                     .withTableKinds(TableMetadata.Kind.REGULAR)
+                                     .withDefaultTypeGen(AbstractTypeGenerators.builder()
+                                                                               .withoutEmpty()
+                                                                               .withUserTypeKeyspace(KEYSPACE)
+                                                                               .withMaxDepth(2)
+                                                                               .withDefaultSetKey(nonEmptyNoDuration)
+                                                                               .build())
+                                     .withPartitionColumnsCount(1)
+                                     .withPrimaryColumnTypeGen(new TypeGenBuilder(nonEmptyNoDuration).withMaxDepth(2).build())
+                                     .withClusteringColumnsBetween(1, 2)
```

Comment:
the reason for 1 partition and 1-2 clustering is to keep the size small.. we have a limit of 64k and with  larger sizes its easier to hit that....